lib/cli/src/Cardano/CLI.hs

```diff
@@ -1435,16 +1450,16 @@ transactionSubmitPayloadArgument = argumentT $ mempty
 -- | [--metadata=JSON]
 --
 -- Note: we decode the JSON just so that we can validate more client-side.
-metadataOption :: Parser ApiTxMetadata
+metadataOption :: Parser (Maybe TxMetadataWithSchema)
```

Comment:
Consider changing its type to `Parser TxMetadataWithSchema` and then using [optional](https://hackage.haskell.org/package/base-4.16.1.0/docs/Control-Applicative.html#v:optional) combinator to recover optionality?

Reply:
how would you change 
```haskell
txMetadataReader :: ReadM (Maybe TxMetadataWithSchema)
txMetadataReader = eitherReader (Aeson.eitherDecode' . BL8.pack)
``` ? 

That is the source of the Maybe, should we blow up instead ?

Reply:
I think it's not that  easy 
```haskell
    describe "Tx Metadata JSON option" $ do
        let parse arg = execParserPure defaultPrefs
                (info metadataOption mempty) ["--metadata", arg]
        let md = detailedMetadata
                (TxMetadata (Map.singleton 42 (TxMetaText "hi")))
        let ok ex (Success res) = ex ==  res
            ok _ _ = False
        let err (Failure _) = True
            err _ = False
        mapM_
            (\(desc, arg, tst) -> it desc (parse arg `shouldSatisfy` tst))
            [ ("valid", "{ \"42\": { \"string\": \"hi\" } }", ok (Just md))
            , ("malformed", "testing", err)
            , ("malformed trailing", "{ \"0\": { \"string\": \"\" } } arstneio", err)
            , ("invalid", "{ \"json\": true }", err)
            , ("null 1", "{ \"0\": null }", err)
            , ("null 2", "null", ok Nothing)
            , ("null 3", "{ }", ok (Just (detailedMetadata mempty)))
            ]
```
to pass that null 2 test we cannot rely on TxMetadataWithSchema

Reply:
Can I close this suggestion as a do-nothing, or we can open a ticket to expand the TxMetadataWithSchema to support the `null` value?